packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -114,6 +112,15 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 	) {
 		this.selfAttendee = new SessionClient(clientSessionId);
 		this.attendees.set(clientSessionId, this.selfAttendee);
+
+		// Set a timeout to check for stale connections after 30 seconds
+		setTimeout(() => {
+			for (const client of this.staleConnectionClients) {
+				// Mark the client as disconnected and remove from the stale connection set
+				client.setDisconnected();
+				this.staleConnectionClients.delete(client);
+			}
+		}, 30000);
```

Comment:
important - got test coverage for this?
Should also add comments about new pattern in code and in the PR description